tests/integration/test_mock_router.py

```diff
@@ -0,0 +1,183 @@
+
+    expected = ["4", 9]  # pass str number as a sanity check that it stays a str
+
+    with httpx.Client(base_url=BASE_URL) as client:
+        response = client.get(f"/items/{expected[0]}/sizes/{expected[1]}")
+        assert response is not None
+        body = response.json()
+        assert "expected" in body
+        assert body["expected"] == expected
+
+
+def test_get_with_bad_input(httpx_mock: HTTPXMock):  # noqa: F811
+    """Look for error raised with invalid path variables"""
+    httpx_mock.add_callback(callback=app.handle_request)
+
+    expected = ["pass", "fail"]
+
+    with pytest.raises(HttpException):
+        with httpx.Client(base_url=BASE_URL) as client:
+            client.get(f"/items/{expected[0]}/sizes/{expected[1]}")
+
+
+def test_post_successful(httpx_mock: HTTPXMock):  # noqa: F811
+    """Pass a vanilla POST request.
+
+    Makes sure that the request parameter is correctly passed to the endpoint function.
+    """
+    httpx_mock.add_callback(callback=app.handle_request)
+
+    request_body = {"detail": {"a key": "a value"}}
+    expected = request_body["detail"]
+
+    with httpx.Client(base_url=BASE_URL) as client:
+        response = client.post("/items", json=request_body)
+        assert response is not None
+
+        body = response.json()
+        assert "expected" in body
+        assert body["expected"] == expected
+
+
+def test_post_failure(httpx_mock: HTTPXMock):  # noqa: F811
+    """Cause the endpoint to raise an HttpException.
+
+    Makes sure that endpoint-defined exceptions are passed up as expected when no
+    exception handler is specified.
+    """
+    httpx_mock.add_callback(callback=app.handle_request)
+
+    # cause a failure by omitting the "detail" key that the endpoint looks for
+    with pytest.raises(HttpException):
+        with httpx.Client(base_url=BASE_URL) as client:
+            client.post("/items", json={})
+
+
+def test_post_failure_with_handler(httpx_mock: HTTPXMock):  # noqa: F811
+    """Cause the endpoint to raise an HttpException.
+
+    Makes sure that exceptions are handled with the specified handler.
+    """
+    app.http_exception_handler = http_exception_handler
+    httpx_mock.add_callback(callback=app.handle_request)
+
+    # cause a failure by omitting the "detail" key that the endpoint looks for
+    with httpx.Client(base_url=BASE_URL) as client:
+        client.post("/items", json={})
+
+
+def test_path_and_function_mismatch():
+    """Make sure that we get an error if path variable names and decorated endpoint
+    function parameter names are not identical.
+    """
+
+    # create a new MockRouter so we don't modify 'app'
+    throwaway = MockRouter()
+
+    with pytest.raises(
+        TypeError,
+        match="Path variables for path '/dummy/{pram1}' do not match the function it decorates",
+    ):
+
+        @throwaway.get("/dummy/{pram1}")
+        def dummy(parameter1: int) -> None:
+            """Dummy function with missing type-hint info"""
```

Comment:
Description does not fit

Reply:
Fixed in [29b6eb2](https://github.com/ghga-de/ghga-service-commons/pull/18/commits/29b6eb2fece1c2f3ade6fde56dfbf50a24842af1)